### PR TITLE
configure: Use pg_config to locate the library location too

### PR DESCRIPTION
**Pull request by whitslack**
On some systems (e.g., Gentoo), there can be multiple major versions of the PostgreSQL client library installed in parallel. There exists a separate `pg_config` and `libpq.so*` for each major version. This Pull Request adds support to C-Lightning's build system to allow the user to choose a specific version of the PostgreSQL client library to build and link against.

**Summary of proposed changes:**

* Introduce a `POSTGRES_LDLIBS` config variable, analogous to the existing `POSTGRES_INCLUDE` variable, and use it in place of `-lpq`, both in `Makefile` and in the Configurator script.
* When running `configure`, allow the user to override the name of `pg_config` by setting the `PG_CONFIG` environment variable. If unset, the default continues to be `pg_config`, which preserves the existing behavior.
* The above changes have the happy side-effect that setting `PG_CONFIG` to an empty string will now cause Configurator's test for PostgreSQL to fail, so C-Lightning will be built without PostgreSQL support, even if a PostgreSQL client library is present on the build system. (The Gentoo ebuild has been hacking the Configurator test code by inserting an `#error` directive to force the test to fail when the user has requested that C-Lightning not support PostgreSQL. That hack is no longer necessary with the changes in this PR.)

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -230,8 +230,10 @@ fi
 require 'python3-mako' "You need the mako module for python3: see doc/INSTALL.md" python3 -c 'import mako'
 
 POSTGRES_INCLUDE=""
-if command -v pg_config 2> /dev/null; then
-    POSTGRES_INCLUDE="-I$(pg_config --includedir)"
+POSTGRES_LDLIBS=""
+if command -v "${PG_CONFIG-pg_config}" >/dev/null; then
```
Comment (whitslack) [inline configure:234]: Note, I fixed the redirection here. (`command -v` prints to stdout, not stderr.)

File: configure [hunk 2]
```diff
@@ -230,8 +230,10 @@ fi
...
+    POSTGRES_INCLUDE="-I$("${PG_CONFIG-pg_config}" --includedir)"
+    POSTGRES_LDLIBS="-L$("${PG_CONFIG-pg_config}" --libdir) -lpq"
```
Comment (cdecker) [inline configure:236]: We could set the `$PG_CONFIG` at the top of the script to avoid having to use the get-or-default construct here.

**Review by cdecker — COMMENTED** [review]
> * The above changes have the happy side-effect that setting `PG_CONFIG` to an empty string will now cause Configurator's test for PostgreSQL to fail, so C-Lightning will be built without PostgreSQL support, even if a PostgreSQL client library is present on the build system. (The Gentoo ebuild has been hacking the Configurator test code by inserting an `#error` directive to force the test to fail when the user has requested that C-Lightning not support PostgreSQL. That hack is no longer necessary with the changes in this PR.)

This applies only to users explicitly setting it to an empty string, not the default behavior, correct? I.e., if the user has the libs installed and doesn't customize the `PG_CONFIG` envvar, then postgresql support will be built in. Just want to make sure that that's still the default behavior :-)

**Comment by whitslack** [comment]
> This applies only to users explicitly setting it to an empty string, not the default behavior, correct? I.e., if the user has the libs installed and doesn't customize the `PG_CONFIG` envvar, then postgresql support will be built in. Just want to make sure that that's still the default behavior :-)

You have to explicitly set it to empty if you want to forcibly disable PostgreSQL support even though you have the PostgreSQL library installed. If you leave `PG_CONFIG` unset, then you'd get the same automagic detection that the existing script implements.

**Comment by whitslack** [comment]
(Actually you can set `PG_CONFIG` to any nonexistent path name to disable PostgreSQL support. It just so happens that `command -v ''` fails too, so empty is a useful value for this purpose.)

File: configure (hunk 2 above)
Comment (whitslack) [reply configure:236]: I've already left the computer for the night (going to bed), but I'll make this change within the day.

**Comment by cdecker** [comment]
> You have to explicitly set it to empty if you want to forcibly disable PostgreSQL support even though you have the PostgreSQL library installed. If you leave `PG_CONFIG` unset, then you'd get the same automagic detection that the existing script implements.

Excellent, thanks for putting my mind at ease :-)

File: configure (hunk 2 above)
Comment (whitslack) [reply configure]: [Implemented.](https://github.com/ElementsProject/lightning/pull/3995/commits/a29632dfbf4353b4868489a9a5e324e3da89a8e6)

**Review by rustyrussell — APPROVED** [review]
Ack a29632dfbf4353b4868489a9a5e324e3da89a8e6

**Pull request merged**